src/spatialdata/_io/format.py

```diff
@@ -149,6 +198,62 @@ def validate_table(
 
 
 CurrentRasterFormat = RasterFormatV01
-CurrentShapesFormat = ShapesFormatV01
+CurrentShapesFormat = ShapesFormatV02
 CurrentPointsFormat = PointsFormatV01
 CurrentTablesFormat = TablesFormatV01
+
+ShapesFormats = {
+    "0.1": ShapesFormatV01(),
+    "0.2": ShapesFormatV02(),
+}
+PointsFormats = {
+    "0.1": PointsFormatV01(),
+}
+TablesFormats = {
+    "0.1": TablesFormatV01(),
+}
+RasterFormats = {
+    "0.1": RasterFormatV01(),
+}
+
+
+def _parse_formats(formats: SpatialDataFormat | list[SpatialDataFormat] | None) -> dict[str, SpatialDataFormat]:
+    parsed = {
+        "raster": CurrentRasterFormat(),
+        "shapes": CurrentShapesFormat(),
+        "points": CurrentPointsFormat(),
+        "tables": CurrentTablesFormat(),
+    }
+    if formats is None:
+        return parsed
+    if not isinstance(formats, list):
+        formats = [formats]
+
+    modified = {
```

Comment:
what does this modified do?

Reply:
It is to check that no duplicate formats are passed by the user as the list of format used for writing. I added a comment.